cla_backend/apps/diagnosis/graph.py

```diff
@@ -137,88 +137,97 @@ def _get_id_default_dict_for(attr, attr_name="attr.name", element="node"):
-            return value
-
         def str_to_bool(s):
             return s.lower() == "true"
 
         # looping through the nodes
         nodes = self.xpath_ns(self.doc, "//ns:node")
         for node in nodes:
             node_id = node.attrib["id"]
-            permanent_node_id = _get_node_data_value_or_default(node, self.KEY_PERMANENT_ID)
+            permanent_node_id = self._get_node_data_value_or_default(
+                node, self.KEY_PERMANENT_ID, is_templated=is_templated
+            )
             node_id_map[node_id] = permanent_node_id
 
             try:
-                order = int(_get_node_data_value_or_default(node, self.KEY_ORDER))
+                order = int(self._get_node_data_value_or_default(node, self.KEY_ORDER, is_templated=is_templated))
             except TypeError:
                 order = 9999
 
-            label = _get_node_data_value_or_default(node, self.KEY_BODY)
+            label = self._get_node_data_value_or_default(node, self.KEY_BODY, is_templated=is_templated)
             if label:
-                label = _get_markdown(label)
+                label = self._get_markdown(label, is_templated)
 
-            help_text = _get_node_data_value_or_default(node, self.KEY_HELP)
+            help_text = self._get_node_data_value_or_default(node, self.KEY_HELP, is_templated=is_templated)
             if help_text:
-                help_text = _get_markdown(help_text)
+                help_text = self._get_markdown(help_text, is_templated=is_templated)
 
-            description = _get_node_data_value_or_default(node, self.KEY_DESCRIPTION)
+            description = self._get_node_data_value_or_default(node, self.KEY_DESCRIPTION, is_templated=is_templated)
             if description:
-                description = _get_markdown(description)
+                description = self._get_markdown(description, is_templated=is_templated)
 
             self.graph.add_node(
                 permanent_node_id,
                 label=label,
-                title=_get_node_data_value_or_default(node, self.KEY_TITLE),
+                title=self._get_node_data_value_or_default(node, self.KEY_TITLE, is_templated=is_templated),
                 description=description,
-                key=_get_node_data_value_or_default(node, self.KEY_BODY, translate=False),
+                key=self._get_node_data_value_or_default(
+                    node, self.KEY_BODY, is_templated=is_templated, translate=False
+                ),
                 order=order,
-                context=_process_context(node),
+                context=self._process_context(node, context_key=context_key, is_templated=is_templated),
                 help_text=help_text,
-                heading=_get_node_data_value_or_default(node, self.KEY_HEADING),
-                data_safety=_get_node_data_value_or_default(node, self.KEY_DATA_SAFETY, str_to_bool),
+                heading=self._get_node_data_value_or_default(node, self.KEY_HEADING, is_templated),
```

Comment:
is_templated needs to be passed as a keyword argument as it is the fifth argument

Reply:
Great find, missed that one. Thank you